DTAdmin/Subjects/AddNewRecordViewController.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  AddNewRecordViewController.swift
+//  DTAdmin
+//
+//  Created by ITA student on 10/18/17.
+//  Copyright © 2017 if-ios-077. All rights reserved.
+//
+
+import UIKit
+
+class AddNewRecordViewController: UIViewController {
+
+    private var httpStatusCode : Int? = nil
+    
+    let queryService = QueryService()
+    
+    @IBOutlet weak var subjectName: UITextField!
+    
+    @IBOutlet weak var subjectDescription: UITextView!
+    
+    var updateDates = false
+    
+    var subjectId : String = ""
+    var name : String = ""
+    var desc : String = ""
+    
+    private func showMessage(message: String){
+        let alert = UIAlertController(title: "Warning", message: message, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    @IBAction func saveNewRecord(_ sender: UIButton) {
+        guard let name = subjectName.text?.capitalized else {return}
+        guard let description = subjectDescription.text?.capitalized else {return}
+        
+        if !name.isEmpty && !description.isEmpty {
+            if !updateDates {
+                queryService.postRequests(parameters : ["subject_name" : name, "subject_description" : description], sufix : "subject/InsertData", completion: {(results:Int?) in
+                    if let code = results {
+                        self.httpStatusCode = code
+                        DispatchQueue.main.async {
+                            if self.httpStatusCode == 200 {
```

Comment:
Why just not use `code` instead of `self.httpStatusCode` ?

Reply:
Because main queue hasn't the variable code. Because results is optional variable I made Optional Binding and if this variable isn't nil I appropriate variable  self.httpStatusCode that value and in a main queue I check the result